src/Tasks/Microsoft.NET.Build.Tasks/PreprocessPackageDependenciesDesignTime.cs

```diff
@@ -182,6 +191,82 @@ private void PopulateAssemblies()
             }
         }
 
+        /// <summary>
+        /// Update FileDefinitions and FileDependencies to pretend that certain Reference items
+        /// explicitly added by a .targets file in the NETStandard.Library are actually normal
+        /// package items. This allows them to show up properly under the "SDK" node in Solution
+        /// Explorer, rather than just under the "Assemblies" node.
+        ///
+        /// This is not meant to be a general mechanism for injecting files that can't be handled
+        /// through the normal NuGet package mechanisms. For that, package authors should be
+        /// creating FileDefinition and FileDependency items instead of Reference items.
+        /// </summary>
+        private void PopulatePreResolvedNetStandardAssemblies()
+        {
+            var preResolvedAssemblyDependencies = new List<ITaskItem>();
+            foreach (var reference in References)
+            {
+                var packageName = reference.GetMetadata("NuGetPackageId");
+                var packageVersion = reference.GetMetadata("NuGetPackageVersion");
+
+                // This is not a "pre-resolved" assembly; skip it.
+                if (packageName == null || packageVersion == null)
+                {
+                    continue;
+                }
+
+                // This is meant only as a workaround for an issue in NETStandard.Library 2.0.*.
+                // Other packages should use the more general mechanism of creating
+                // FileDefinition and FileDependency items rather than Reference items.
+                if (packageName != "NETStandard.Library")
+                {
+                    continue;
+                }
```

Comment:
@nguerrera @srivatsn Conceptually, we need three pieces of metadata to properly determine where a `Reference` should end up: the package name, package version, and target framework. The last is necessary for multi-targeting scenarios. The NETStandard.Library package provides `NuGetPackageId` and `NuGetPackageVersion` but not the target framework. So no matter what we do we will need to have some NETStandard.Library-specific workaround to inject that info.

I see a couple of issues with turning this into a general mechanism:

1. We need to extend it to other item types. Definitely `AnalyzerReference`, but what about other item types?
2. Specifying the target framework correctly might be tricky for package authors.